src/client/subscriptions.rs

```diff
@@ -386,13 +391,17 @@ impl Client {
     }
 
     /// Add and edit price intervals on a subscription.
-    pub async fn price_intervals(&self, id: &str, params: &PriceIntervalsRequest) -> Result<Subscription, Error> {
-        let req = self.build_request(
+    pub async fn price_intervals(&self, id: &str, params: &PriceIntervalsRequest<'_>) -> Result<Subscription, Error> {
+        let mut req = self.build_request(
             Method::POST,
             SUBSCRIPTIONS_PATH
             .chain_one(id)
             .chain_one("price_intervals")
         );
+        if let Some(key) = params.idempotency_key {
+            req = req.header("Idempotency-Key", key);
+        }
```

Comment:
how do you make sure that the 2nd request has the same idempotency key as the first request? First request needs to output the idempotency key right?

Reply:
I think you just generate one ahead of time and use it for all the retries.